subprojects/messaging/src/main/java/org/gradle/internal/serialize/ExceptionPlaceholder.java

```diff
@@ -206,6 +222,53 @@ private void diagnoseNullPointerException(NullPointerException e) {
         }
     }
 
+    private static Method findCandidateGetCausesMethod(Throwable throwable) {
+        Method[] declaredMethods = throwable.getClass().getDeclaredMethods();
+        for (Method method : declaredMethods) {
+            if (CANDIDATE_GET_CAUSES.contains(method.getName())) {
+                Class<?> returnType = method.getReturnType();
+                if (Collection.class.isAssignableFrom(returnType)) {
```

Comment:
Maybe use `Iterable` instead of `Collection`?

Reply:
I think it's fine as is. Sure we can make it more generic but I doubt we'd find a real use case. If we did we can revisit.